config/config.go

```diff
@@ -115,6 +119,17 @@ func (c *Config) Entry(name string) (*Entry, error) {
 	return entry, nil
 }
 
+// DeleteEntry deletes Entry object named `name`
+func (c *Config) DeleteEntry(name string) error {
+	_, exist := c.Entries[name]
+	if !exist {
+		return fmt.Errorf("config `%s` does not exist", name)
+	}
+
+	delete(c.Entries, name)
```

Comment:
I think if config is `current`, we should not delete it.
https://github.com/screwdriver-cd/screwdriver/issues/2050#issuecomment-607713836

Reply:
I forge the design
> he config to which current are pointing can't be deleted by sd-local config delete <config-name>